homeassistant/components/watchyourlan/sensor.py

```diff
@@ -0,0 +1,134 @@
+"""Support for the WatchYourLAN service."""
+
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .coordinator import WatchYourLANUpdateCoordinator
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the WatchYourLAN sensors."""
+    coordinator = WatchYourLANUpdateCoordinator(hass, entry.data)
+    await coordinator.async_config_entry_first_refresh()
```

Comment:
this shound be moved to the init. The coordinator can be stored in entry.runtime_data